### ISSUE TITLE
Flaky failures of "corelib_2/bigint_from_test" in dartkb

### ISSUE THREAD

**Issue by sjindel-google** [dart-lang/sdk#34901]
```
FAILED: dartkb-vm debug_x64 corelib_2/bigint_from_test
Expected: Pass
Actual: RuntimeError

--- Command "vm_compile_to_kernel" (took 01.000657s):
DART_CONFIGURATION=DebugX64 /b/s/w/ir/pkg/vm/tool/gen_kernel --no-aot --platform=out/DebugX64/vm_platform_strong.dill -o /b/s/w/ir/out/DebugX64/generated_compilations/dartkb/tests_corelib_2_bigint_from_test/out.dill /b/s/w/ir/tests/corelib_2/bigint_from_test.dart --packages=/b/s/w/ir/.packages -Ddart.developer.causal_async_stacks=true --gen-bytecode --drop-ast

exit code:
0

--- Command "vm" (took 02.000254s):
DART_CONFIGURATION=DebugX64 out/DebugX64/dart --enable_interpreter --sync-async --ignore-unrecognized-flags --packages=/b/s/w/ir/.packages /b/s/w/ir/out/DebugX64/generated_compilations/dartkb/tests_corelib_2_bigint_from_test/out.dill

exit code:
255

stderr:
Unhandled exception:
Expect.equals(expected: <-4294967296>, actual: <-4316718722>) fails.
#0      Expect._fail (package:expect/expect.dart)
#1      Expect.equals (package:expect/expect.dart)
#2      testInt (file:///b/s/w/ir/tests/corelib_2/bigint_from_test.dart)
#3      main (file:///b/s/w/ir/tests/corelib_2/bigint_from_test.dart)
#4      _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart)
#5      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart)

--- Re-run this test:
python tools/test.py -m debug -c dartkb -n dartkb-mixed-linux-debug-x64 corelib_2/bigint_from_test
```

**Comment by alexmarkov** [comment]
The test corelib_2/bigint_test fails for the same reason more reliably.

These tests fail in mixed mode (with --enable-interpreter) due to switching from non-intrinsic implementation of BigInt methods to intrinsics on the fly: when BigInt code starts running it is interpreted and intrinsics are not used, for example _estimateQuotientDigit returns 1. Meanwhile, _mulAdd could be compiled and it uses intrinsic implementation which would process 2 digits. BigInt implementation does not account for switching to intrinsic implementation on the fly and mixing intrinsic and non-intrinsic implementations of BigInt methods.

If Dart asserts are enabled, these tests fail on assertions https://github.com/dart-lang/sdk/blob/d91e0ad3d793b43b43e069e330f9272df4ba5561/runtime/lib/bigint_patch.dart#L1416 and https://github.com/dart-lang/sdk/blob/d91e0ad3d793b43b43e069e330f9272df4ba5561/runtime/lib/bigint_patch.dart#L2655

@crelier could you please take a look?

**Comment by crelier** [comment]
In order to verify a fix for this issue, I'd like to reproduce it first, but I am not able to get this assert fault, either for bigint_test or bigint_from_test.
I sometimes get a timeout, or an exhausted heap, but never an assert fault. I suppose I am not enabling the asserts the correct way?
`python tools/test.py -m debug -c dartkb  --vm-options "--enable-interpreter --enable-asserts" corelib_2/bigint_test
`

**Comment by alexmarkov** [comment]
I can reliably reproduce assertion using your command (after removing .status file entries) if I build platform with bytecode:
```
rm -rf out
tools/gn.py -m debug,release -a x64 --bytecode
tools/build.py -m debug,release -a x64 runtime
```

**Comment by crelier** [comment]
Ah, of course, my platform was missing bytecode. Thanks!

**Comment by crelier** [comment]
Fixed by [7de2510](https://dart.googlesource.com/sdk/+/7de25103398461890acb164b4a8efdaa9da8c89f)
